lib/trivial.nix

```diff
@@ -92,11 +92,16 @@ rec {
 
   ## nixpks version strings
 
+  release-info = import ../release-info.nix;
+
   # The current full nixpkgs version number.
   version = release + versionSuffix;
 
   # The current nixpkgs version number as string.
-  release = lib.strings.fileContents ../.version;
+  release = release-info.release;
+
+  # The current nixpkgs version code name.
+  codeName = release-info.codename;
```

Comment:
Why not just put the release info here? I don't really see the point in putting it in a separate file (except for easy parseability, but that's already lost if you use a Nix expression instead of a plain file like .version).

Reply:
In terms of data locality, the release and code name get updated at the same time so it's best to keep them together.

In terms of discoverability, `lib/trivial.nix` is a misc util library. I don't have a strong opinion where the information should be except that `lib/trivial.nix` doesn't seem like the right place. I can see myself running `cat release-info.nix` on a channel or checkout to quickly figure out where I stand.

In terms of parse-ability, Nix is not too hard to parse. JSON would be even easier but doesn't allow comments.

Reply:
fixed in 49a9011